test/Microsoft.AspNetCore.Mvc.Razor.Test/TagHelpers/TagHelperComponentTagHelperTest.cs

```diff
@@ -0,0 +1,291 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Logging.Testing;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.TagHelpers
+{
+    public class TagHelperComponentTagHelperTest
+    {
+        [Fact]
+        public void Initialized_InOrder()
+        {
+            // Arrange            
+            var tagHelperContext = new TagHelperContext(
+                "head",
+                allAttributes: new TagHelperAttributeList(
+                    Enumerable.Empty<TagHelperAttribute>()),
+                items: new Dictionary<object, object>(),
+                uniqueId: "test");
+
+            var incrementer = 0;
+            var testTagHelperComponentTagHelper = new TestTagHelperComponentTagHelper(new List<ITagHelperComponent>()
+            {
+                new CallbackTagHelperComponent(
```

Comment:
Put these in a mixed up order so we ensure that the component is actually ordering. As it stands currently they'll be called in the correct order even if the code didn't order them. Should do the same in the order ordering test case.